source/extended-json.rst

```diff
@@ -0,0 +1,618 @@
+values are encoded as follows:
+
++----------------------------------------+----------------------------------------+
+|**Value**                               |**String**                              |
++----------------------------------------+----------------------------------------+
+|Positive Infinity                       |``Infinity``                            |
++----------------------------------------+----------------------------------------+
+|Negative Infinity                       |``-Infinity``                           |
++----------------------------------------+----------------------------------------+
+|NaN (all variants)                      |``NaN``                                 |
++----------------------------------------+----------------------------------------+
+
+.. _extended JSON format for the Decimal128 type: https://github.com/mongodb/specifications/blob/master/source/bson-decimal128/decimal128.rst#to-string-representation
+
+For example, a BSON floating-point number with a value of positive infinity
+would be encoded as extended JSON as follows::
+
+  {"$numberDouble": "NaN"}
+
+Option Names
+------------
+
+Parsers and generators MAY be configurable to handle both Canonical and
+Historical Extended JSON formats. In such a case, the following option names MAY
+be used to distinguish between these two formats:
+
+* ``canonicalExtendedJSON`` (references Extended JSON as described in this specification)
+* ``legacyExtendedJSON`` (references Historical Extended JSON)
+
+Deviating from these names is perfectly acceptable as described in similar
+specifications. These names may also be used as "base names" by appending or
+prepending as appropriate to form method names and other option names. This
+specification does not require any specific options to be made available.
+
+Example
+=======
+
+Input Document
+--------------
+
+Consider the following document, written in Groovy with the MongoDB Java Driver::
+
+  {
+    "_id": new ObjectId("57e193d7a9cc81b4027498b5"),
+    "Symbol": new BsonSymbol("symbol"),
+    "String": "string",
+    "Int32": 42,
+    "Int64": 42L,
+    "Double": 42.42,
+    "Binary": UUID.fromString("c8edabc3-f738-4ca3-b68d-ab92a91478a3"),
+    "BinaryUserDefined": new Binary((byte) 0x80, new byte[]{1, 2, 3, 4, 5}),
+    "Code": new Code("function() {}"),
+    "CodeWithScope": new CodeWithScope("function() {}", new Document()),
+    "Subdocument": new Document("foo", "bar"),
+    "Array": Arrays.asList(1, 2, 3, 4, 5),
+    "Timestamp": new BSONTimestamp(42, 1),
+    "Regex": new BsonRegularExpression("pattern"),
+    "DatetimeEpoch": new Date(0),
+    "DatetimePositive": new Date(Long.MAX_VALUE),
+    "DatetimeNegative": new Date(Long.MIN_VALUE),
+    "True": true,
+    "False": false,
+    "DBPointer": new BsonDbPointer(
+         "db.collection", new ObjectId("57e193d7a9cc81b4027498b1")),
+    "DBRef": new DBRef(
+          "database", "collection", new ObjectId("57fd71e96e32ab4225b723fb")),
```

Comment:
Should this be broken out into two examples with and without a database name, as is done for Binary and DateTime variations above? I'll assume that the DBRef class in the driver doesn't support extra fields beyond the `$`-prefixed three, so we can skip that example.